src/commands/gitHub/disconnectRepo/DisconnectRepositoryConfirmStep.ts

```diff
@@ -0,0 +1,25 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { gitHubUrlParse } from "@microsoft/vscode-azext-github";
+import { AzureWizardPromptStep, nonNullValueAndProp } from "@microsoft/vscode-azext-utils";
+import { localize } from "../../../utils/localize";
+import type { IDisconnectRepoContext } from "./IDisconnectRepoContext";
+
+export class DisconnectRepositoryConfirmStep extends AzureWizardPromptStep<IDisconnectRepoContext> {
+    public async prompt(context: IDisconnectRepoContext): Promise<void> {
+        const { ownerOrOrganization: owner, repositoryName: repo } = gitHubUrlParse(nonNullValueAndProp(context.sourceControl, 'repoUrl'));
+
+        await context.ui.showWarningMessage(
+            localize('disconnectRepositoryWarning', 'Disconnect from GitHub repository "{0}"? \n\nThis will not affect your container app\'s active deployment. You may reconnect a repository at any time.', `${owner}/${repo}`),
```

Comment:
I don't actually know what's right, but is this the correct version of effect/affect to use here? I swear I took grammar classes in school...

Reply:
Yeah I'm pretty sure this is the right version to use.  The way I always remembered it was like, `affect` was usually acting upon a noun, and `effect` described a noun.  So like... light `effect` vs. this `affects` that.

Reply:
Matt's right, though people usually just say effect is the noun and affect is the verb lol.

Reply:
That makes more sense lol